src/screens/FeedScreen/Header.js

```diff
@@ -102,15 +114,40 @@ const _renderAnonimity = ({
   height,
   headerStyle,
   showAnonymousOption = false,
-  onHeaderOptionClicked = () => {},
-  hideThreeDot,
   version = 1,
   anonUserInfo = {},
   isPostDetail,
   karmaScore = 0,
-  isBlurredPost = false
+  isBlurredPost = false,
+  isFollow = false,
+  onPressFollUnFoll = () => {},
+  onDeletePost = () => {},
+  isShowDelete = false,
+  isSelf = false,
+  hideThreeDot
 }) => {
   const navigation = useNavigation();
+  const refSheet = React.useRef();
+  const dataSheet = [
+    {
+      id: 1,
+      name: 'Share link',
+      icon: <ShareAndroidIcon />
+    }
+  ];
+
+  if (isShowDelete) {
+    dataSheet.push({
+      id: 2,
+      name: 'Delete post',
+      icon: <TrashRed />,
+      onPress: () => {
+        refSheet.current.close();
+        onDeletePost();
+      },
+      style: {color: colors.red}
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `dataSheet` array is being mutated directly using the `push` method. This can lead to unexpected behavior and bugs in React components. Instead, consider creating a new array with the additional object when needed.
```diff
- if (isShowDelete) {
-   dataSheet.push({
-     id: 2,
-     name: 'Delete post',
-     icon: <TrashRed />,
-     onPress: () => {
-       refSheet.current.close();
-       onDeletePost();
-     },
-     style: {color: colors.red}
-   });
- }
+ const deleteOption = isShowDelete ? [{
+   id: 2,
+   name: 'Delete post',
+   icon: <TrashRed />,
+   onPress: () => {
+     refSheet.current.close();
+     onDeletePost();
+   },
+   style: {color: colors.red}
+ }] : [];
+ 
+ const finalDataSheet = [...dataSheet, ...deleteOption];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->